modules/preview-react/segmented-control/lib/SegmentedControlItem.tsx

```diff
@@ -0,0 +1,171 @@
+   * If this property is not provided, it will default to a string representation
+   * of the the zero-based index of the item when it was initialized.
+   */
+  'data-id'?: string;
+  /**
+   * Optional id. If not set, it will inherit the ID passed to the `SegmentedControl` component and append the
+   * index at the end. Only set this for advanced cases.
+   */
+  id?: string;
+  /**
+   * Part of the ARIA specification for buttons. Lets screen readers know which button is active. This
+   * should either be `true` or `false`. This is automatically set by the
+   * component and should only be used in advanced cases.
+   */
+  'aria-pressed'?: boolean;
+  /**
+   * The icon of the button.
+   */
+  icon?: CanvasSystemIcon;
+  /**
+   * Tooltip Props
+   */
+  tooltipProps?: Omit<TooltipProps, 'children'>;
+}
+
+const getButtonSize = (size: ButtonContainerProps['size']) => {
+  switch (size) {
+    case 'large':
+      return 'medium';
+    case 'medium':
+      return 'small';
+    case 'small':
+      return 'extraSmall';
+    default:
+      return 'medium';
+  }
+};
+
+const getIconButtonColors = (toggled?: boolean): ButtonColors => {
+  return {
+    default: {
+      background: toggled ? colors.frenchVanilla100 : colors.soap200,
+      border: toggled ? colors.licorice200 : 'transparent',
+      icon: toggled ? colors.blackPepper400 : colors.licorice400,
+      label: toggled ? colors.blackPepper400 : colors.licorice400,
+    },
+    hover: {
+      background: toggled ? colors.frenchVanilla100 : colors.soap400,
+      icon: colors.licorice400,
+      label: colors.licorice400,
+    },
+    active: {},
+    focus: {},
+    disabled: {
+      background: colors.soap200,
+      opacity: '1',
+    },
+  };
+};
+
+const getPaddingStyles = (
+  children: React.ReactNode,
+  size: ButtonContainerProps['size'],
+  icon: CanvasSystemIcon | undefined
+) => {
+  if (!children) {
+    return 0;
+  }
+
+  switch (size) {
+    case 'large':
+      return icon ? `0 ${space.m} 0 20px` : `0 ${space.m}`;
+
+    case 'medium':
+      return icon ? `0 20px 0 ${space.s}` : `0 ${space.s}`;
+
+    case 'small':
+      return icon ? `0 ${space.xs} 0 ${space.xxs}` : `0 ${space.xs}`;
+
+    default:
+      return icon ? `0 20px 0 ${space.s}` : `0 ${space.s}`;
+  }
+};
+
+const geButtonStyles = (size: ButtonSizes, children: React.ReactNode, icon?: CanvasSystemIcon) => {
+  const buttonSize = getButtonSize(size);
+  const minWidthValue = getMinWidthStyles(children, children ? size : buttonSize);
+
+  return {
+    height: getMinWidthStyles(false, buttonSize),
+    minWidth: minWidthValue,
+    padding: getPaddingStyles(children, size, icon),
+  };
+};
+
+const Container = ({
+  tooltipProps,
+  children,
+}: {
+  tooltipProps?: Omit<TooltipProps, 'children'>;
+  children: React.ReactElement;
+}) => {
+  return tooltipProps ? (
+    <Tooltip title={tooltipProps.title}>{children}</Tooltip>
```

Comment:
This is a little strange to me. We say that `tooltipProps` is all of `TooltipProps` minus `children`, but we only forward `title`? Is there a reason we don't forward all the props?

```suggestion
    <Tooltip {...tooltipProps}>{children}</Tooltip>
```

Reply:
I'm also curious about overflow tooltip. Should we use `OverflowTooltip` if no `title` is provided? What happens if the contents of a button get truncated?